### PR TITLE
LPS-181083

### PR DESCRIPTION
**Pull request by CaiocSiqueira**
https://issues.liferay.com/browse/LPS-181083

PASSED Testing on: https://github.com/CaiocSiqueira/liferay-portal/pull/36

### CODE REVIEW

**Comment by CaiocSiqueira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 1]
```diff
@@ -2154,6 +2154,181 @@ definition {
...
+
+		task ("Add sort widget to the search page and perform a search") {
+			Search.openSearchPage();
+
+			SearchPortlets.addWidgets(searchPortletList = "Sort");
+
+			Search.openSearchPage(searchTerm = "fruit");
+		}
+
+		task ("Edit sort widget configuration") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2252]: Rework this entire task block by doing the following:
- For adding a sort option, create and use a macro called SearchPortlets.addSortOption that accepts a variable for each field of the sort option
- Don't use for loops for this
- Add a task for each sort option added giving a short description of what you are sorting by

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 2]
```diff
@@ -2154,6 +2154,181 @@ definition {
...
+				var count = ${count} + 1;
+
+				ProductMenu.gotoPortlet(
+					category = "Content & Data",
+					portlet = "Custom Objects");
+			}
+		}
+
+		task ("Add sort widget to the search page and perform a search") {
+			Search.openSearchPage();
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2245]: delete this as it's redundant

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 3]
```diff
@@ -2154,6 +2154,181 @@ definition {
...
+
+			SearchPortlets.addWidgets(searchPortletList = "Sort");
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2247]: use JSONLayout.addWidgetToPublicLayout

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 4]
```diff
@@ -2154,6 +2154,181 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-172131"
+	@priority = 3
+	test SortByCustomObjectFields {
+		task ("Create object with Title, Date and Site ID fields in it") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2160]: Make the same suggestion for creating the object definition and entries from the review here: https://github.com/joshchong/liferay-portal/pull/701#pullrequestreview-1425248913

**Review by joshchong — CHANGES_REQUESTED** [review]

**Comment by joshchong** [comment]
@CaiocSiqueira forwarding in https://github.com/joshchong/liferay-portal/pull/720. please see changes. thanks!

**Pull request closed without merging**